### PR TITLE
Add support for solidity “constructor” keyword

### PR DESCRIPTION
**Pull request by ChenfengWei0**
Add support for "constructor" keywords in solidity front-end.

### CODE REVIEW

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
Can you add regression tests with assertions that rely on the constructors/destructors?

**Comment by ChenfengWei0** [comment]
> Can you add regression tests with assertions that rely on the constructors/destructors?

Sure

File: regression/esbmc-solidity/constructor_1/test.desc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CORE
+contract.solast
+--function empty --contract contract.sol
+^VERIFICATION SUCCESSFUL$
```
Comment (kunjsong01) [inline regression/esbmc-solidity/constructor_1/test.desc:4]: Could you add a test case for "VERIFICATION FAILED" please?

**Review by kunjsong01 — CHANGES_REQUESTED** [review]

**Comment by ChenfengWei0** [comment]
> Can you add regression tests with assertions that rely on the constructors/destructors?

**Comment by ChenfengWei0** [comment]
I just found that some features of constructor are not fully supported. Likely to take more time to resolve. Sorry.

**Pull request closed without merging**